### PR TITLE
Add a hera proposal process

### PR DESCRIPTION
**Pull request by samj1912**
This PR adds a HEP (Hera Enhancement Proposal) process to introduce new large features into hera.

### CODE REVIEW

File: hep/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Hera Enhancement Proposals (HEP)
+
+This folder is used to manage Hera Enhancement Proposals (HEPs). An enhancement proposal is recommend for any significant change, including new features and enhancements.
```
Comment (elliotgunton) [inline hep/README.md:3]:
```suggestion
This folder is used to manage Hera Enhancement Proposals (HEPs). An enhancement proposal is recommended for any significant change, including new features and enhancements.
```

File: hep/proposals/0000-template.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+# Alternatives
+
+- What other designs have been considered?
+- Why is this proposal the best?
+- What is the impact of not doing this?
+
+# Prior Art
+
+Discuss prior art, both the good and bad.
```
Comment (elliotgunton) [inline hep/proposals/0000-template.md:87]: What's "prior art"? Typo or jargon? 😅

File: hep/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## HEP Process
+
+To get a significant feature into Hera, first, a HEP needs to be approved and merged into the hera repo. Once it is merged, it's considered 'Accepted' and may be 'Implemented' to be included in the project. These steps will get an HEP to be considered:
+
+1. Fork the hera repo: https://github.com/argoproj-labs/hera
+2. Copy 0000-template.md to `proposals/0000-<proposal-name>.md` (where '<proposal-name>' is descriptive.).
```
Comment (elliotgunton) [inline hep/README.md:10]: Disappears in the md
```suggestion
2. Copy 0000-template.md to `proposals/0000-<proposal-name>.md` (where '\<proposal-name>' is descriptive.).
```

File: hep/proposals/0000-template.md (hunk 2 above)
Comment (samj1912) [reply hep/proposals/0000-template.md:87]: > Prior art is any evidence that your invention is already known. Prior art does not need to exist physically or be commercially available. It is enough that someone, somewhere, sometime previously has described or shown or made something that contains a use of technology that is very similar to your invention.

**Review by flaviuvadan — APPROVED** [review]

File: hep/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+2. Copy 0000-template.md to `proposals/0000-<proposal-name>.md` (where `<proposal-name>` is descriptive.).
+3. Fill in the details for the HEP. Any section can be marked as "N/A" if not applicable.
+4. Submit a pull request. The pull request is the time to get review of the proposal from the larger community.
+5. Once the pull request is approved by a maintainer, the HEP will enter the 'Final Comment Period'.
+
+### Final Comment Period
+
+When a HEP enters the FCP the following will happen:
+
+1. A maintainer will apply the "Final Comment Period" label.
```
Comment (flaviuvadan) [inline hep/README.md:19]: Can we automatically apply labels when this folder is modified?

**Review by flaviuvadan — APPROVED** [review]

**Pull request merged**